client/packages/common/src/ui/components/buttons/standard/ButtonWithIcon.tsx

```diff
@@ -13,6 +13,22 @@ export interface ButtonWithIconProps extends ButtonProps {
   disabled?: boolean;
   shrinkThreshold?: 'sm' | 'md' | 'lg' | 'xl';
 }
+const formatLabel = (label: string) => {
+  if (label.indexOf('&') === -1) {
+    return label;
+  }
+  const match = /(.*)\&(.)(.*)/.exec(label);
+  if (!match || match.length !== 4) {
+    return label;
+  }
+  return (
+    <>
+      {match[1]}
+      <u>{match[2]}</u>
+      {match[3]}
+    </>
+  );
+};
```

Comment:
This is a good way to allow formatting of localised strings, but shouldn't this apply to *all* strings, not just Buttons? Also, if we want to open the can of worms that is formatted localisations, should we have a more general solution? (Basic Markdown interpretation would be my pick, rather than inventing a new markup syntax.)

Reply:
that would require a ton of change though! What do you propose for basic markdown? the likes of `_S_` and `**S**`?

Reply:
> that would require a ton of change though! What do you propose for basic markdown? the likes of `_S_` and `**S**`?

Yeah, just those for now, I guess.

As far as making it available globally for all strings, the problem is that it has to return HTML, and the standard "translate" function just returns strings. We could return an additional "format" function from the Language Provider which we would only call for strings that require formatting. So in this case, something like:
```ts
format(t("button.scan")) // returns <span><u>S</u>can</span>
```

We do something like that in Conforma, but then again, we use Markdown in configurations extensively there, so have a dedicated Markdown library for that, which we probably don't need here.

Having said that, is an underlined first letter actually the best way to indicate a shortcut? That seems quite old fashioned to me. I dunno, maybe it's a Windows thing, but I haven't seen that approach for many years. (Plus we don't follow that convention anywhere else, so it seems a bit odd to use it arbitrarily here?) 

That's the advantage of the KBar module -- all the shortcuts are available in a list if you want to find out what they are. It's a shame if that can't handle modifiers properly.